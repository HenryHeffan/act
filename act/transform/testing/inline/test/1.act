function addone (int x) : int
{ 
  chp {
    self := x + 1;
    self := self + 1
  }
}


defproc test ()
{
  int x, y;

  chp {
    x := addone (y)
  }
}

test t;
