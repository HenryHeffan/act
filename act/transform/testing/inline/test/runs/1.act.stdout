defproc test ();
function addone (int<32> x) : int<32>;

defproc test ()
{

/* instances */
int<32> y;
int<32> x;

/* connections */
chp {
x:=y+0x1+0x1
}
}

function addone (int<32> x) : int<32>
{

/* instances */

/* connections */
chp {
self:=x+0x1;self:=self+0x1
}
}


/* instances */
test t;

/* connections */
